test/dst/model.go

```diff
@@ -252,9 +252,10 @@ func (m *Model) ValidateCompletePromise(t int64, req *t_api.Request, res *t_api.
 	switch res.CompletePromise.Status {
 	case t_api.StatusOK:
 		if pm.completed() {
-			if !pm.idempotencyKeyForCompleteMatch(res.CompletePromise.Promise) {
+			if !(!req.CompletePromise.Strict && pm.promise.State == promise.Timedout) &&
```

Comment:
I'm finding this if statement a little hard to follow, so I plugged it into a program that simplifies boolean expressions. To me the result is a little easier to understand, let me know what you think.

```
!(!req.CompletePromise.Strict && pm.promise.State == promise.Timedout)
  && !pm.idempotencyKeyForCompleteMatch(res.CompletePromise.Promise)

a = req.CompletePromise.Strict
b = pm.promise.State == promise.Timedout
c = pm.idempotencyKeyForCompleteMatch(res.CompletePromise.Promise)

!(!a && b) && !c
!c && !(!a && b)  put c first (I find this a little easier to read)

~c ∧ ~(~a ∧ b)    convert to boolean algebra
~c ∧ (a ∨ ~b)     an equivalent expression

!c && (a || !b)   convert back to go code, and finally put back in substitutions

!pm.idempotencyKeyForCompleteMatch(res.CompletePromise.Promise)
  && (req.CompletePromise.Strict || pm.promise.State != promise.Timedout)
```

[Here](https://www.emathhelp.net/en/calculators/discrete-mathematics/boolean-algebra-calculator/?f=%7Ec%E2%88%A7%7E%28%7Ea%E2%88%A7b%29) is the equivalency proof, and here are the corresponding truth tables for both expressions:
- [~c ∧ ~(~a ∧ b)](https://www.emathhelp.net/en/calculators/discrete-mathematics/truth-table-calculator/?f=%7Ec+%E2%88%A7+%7E%28%7Ea+%E2%88%A7+b%29&v=)
- [~c ∧ (a ∨ ~b)](https://www.emathhelp.net/en/calculators/discrete-mathematics/truth-table-calculator/?f=%7Ec+%E2%88%A7+%28a+%E2%88%A8+%7Eb%29&v=)

Reply:
Sweet! Yes, that looks much better.

Thanks for the detailed explanation.